internal/uvm/virtual_device.go

```diff
@@ -11,6 +11,13 @@ import (
 	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
 )
 
+const (
+	VPCIDeviceIDType       = "vpci"
```

Comment:
What are the differences between `vpci` and `vpciLocationPath`?

Reply:
Basically, when a container requests a device, if it is a process isolated container it can only request a device with an IDType of `vpciLocationPath` or `vpciClassGUID` (which will be changed to `class`). `vpci` and `gpu` are for hyper-v isolated containers only, which are still in the process of being fully supported. I can probably take out the `vpci` field for this PR.

Reply:
Do `vpci` and `vpciLocationPath` indicate that different types of values are passed? For instance, I think vpci is used like `vpci://<some instance id>`, what do you pass after `vpciLocationPath://`?

Reply:
`vpci-location-path://<some device location path>` This is only directly used in process isolated containers.

Reply:
Okay, so a location path is distinct from an instance ID?

I would propose we support 5 types here:
- `vpci-instance-id`: The instance ID of the VPCI device.
- `vpci`: Legacy, alias for `vpci-instance-id`.
- `vpci-location-path`: The location path of the VPCI device.
- `vpci-class-id`: The class ID of the VPCI device.
- `class`: Legacy, alias for `vpci-class-id`.

However, you're the expert in this area. So I'm looking to you to help us understand if exposing this set of concepts makes sense. :)

Reply:
Why do you propose the legacy versions? Just to be more specific or are you worried about breaking anything? We don't need to keep `vpci` for example, that's not being used anywhere.

Reply:
[This](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/locationpath-registry-subkey) is the best public doc that explains what a location path is on a device (look specifically at the third paragraph). A good description of device instance ID is [here](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/device-instance-ids). Let me know if that helps or not.

Reply:
I think keeping `class` is good as it's publicly documented. If no one is using `vpci` we can remove it.

Reply:
We've discussed and decided on the following types for now:

	GPUDeviceIDType         = "gpu"
	VPCILocationPathIDType  = "vpci-location-path"
	VPCIClassGUIDTypeLegacy = "class"
	VPCIClassGUIDType       = "vpci-class-guid"